src/UniversalRewardsDistributor.sol

```diff
@@ -4,75 +4,81 @@ pragma solidity 0.8.21;
-///         It is largely inspired by Morpho's current rewards distributor:
-///         https://github.com/morpho-dao/morpho-v1/blob/main/src/common/rewards-distribution/RewardsDistributor.sol
+/// @notice This contract enables the distribution of various reward tokens to multiple accounts using different
+/// permissionless Merkle trees. It is largely inspired by Morpho's current rewards distributor:
+/// https://github.com/morpho-dao/morpho-v1/blob/main/src/common/rewards-distribution/RewardsDistributor.sol
 contract UniversalRewardsDistributor is IUniversalRewardsDistributor {
     using SafeTransferLib for ERC20;
 
-    /// @notice The merkle tree's roots of a given distribution.
+    /* STORAGE */
+
+    /// @notice The merkle tree's root of a given distribution.
     bytes32 public root;
 
     /// @notice The optional ipfs hash containing metadata about the root (e.g. the merkle tree itself).
     bytes32 public ipfsHash;
 
     /// @notice The `amount` of `reward` token already claimed by `account` for one given distribution.
-    mapping(address => mapping(address => uint256)) public claimed;
+    mapping(address account => mapping(address reward => uint256 amount)) public claimed;
 
     /// @notice The address that can update the distribution parameters, and freeze a root.
     address public owner;
 
     /// @notice The addresses that can update the merkle tree's root for a given distribution.
     mapping(address => bool) public isUpdater;
 
-    /// @notice The timelock before a root update
+    /// @notice The timelock before a root update.
     uint256 public timelock;
 
     /// @notice The pending root of the distribution.
     /// @dev If the pending root is set, the root can be updated after the timelock has expired.
     /// @dev The pending root is skipped if the timelock is set to 0.
     PendingRoot public pendingRoot;
 
-    modifier onlyOwnerOrUpdater() {
-        require(isUpdater[msg.sender] || msg.sender == owner, ErrorsLib.CALLER_NOT_OWNER_OR_UPDATER);
-        _;
-    }
+    /* MODIFIERS */
 
+    /// @notice Reverts if the caller is not the owner.
     modifier onlyOwner() {
         require(msg.sender == owner, ErrorsLib.CALLER_NOT_OWNER);
         _;
     }
 
+    /// @notice Reverts if the caller is not the owner nor an updater.
+    modifier onlyOwnerOrUpdater() {
+        require(isUpdater[msg.sender] || msg.sender == owner, ErrorsLib.CALLER_NOT_OWNER_OR_UPDATER);
+        _;
+    }
+
+    /* CONSTRUCTOR */
+
     /// @notice Initializes the contract.
     /// @param initialOwner The initial owner of the contract.
     /// @param initialTimelock The initial timelock of the contract.
     /// @param initialRoot The initial merkle tree's root.
     /// @param initialIpfsHash The optional ipfs hash containing metadata about the root (e.g. the merkle tree itself).
+    /// @dev Warning: The `initialIpfsHash` might not correspond to the `initialRoot`.
```

Comment:
this is not a requirement. If we want to add a comment about it, I prefer to say what the ipfsHash should describe instead of what it should not describe (infinite list)

Reply:
Yes it's not a requirements so better to write no? It's 100% sure that we'll have this finding in audits

Reply:
They are not that independent as the primary goal of the ipfs hash is to be able to recompute the root.

"The desc of the params", not sure to understand what it means though.

We can remove it but I still think that a comment saying that the contract does not enforce the metadata to match the root would useful

Reply:
No big deal i'm ok with the comment